api/src/wfl/service/firecloud.clj

```diff
@@ -126,6 +127,81 @@
                               {:Content/type "text/tab-separated-values"
                                :entities     (slurp file)})})))
 
+(defn import-entity-set
+  "
+  Formats existing sample ENTITIES into a Terra-compatible tsv
+  and uploads to WORKSPACE as an entity set entry identified by SET-NAME.
+
+  All entity references should correspond to existing entities.
+  All entity references should be of the same type.
+
+  Parameters
+  ----------
+    workspace - Fully-qualified Terra Workspace in which the entity set
+                will be created
+    set-name  - Unique identifier for the entity set
+    entities  - List of entity [Type Name] pairs
+
+  Example
+  -------
+    (import-entity-set \"workspace-namespace/workspace-name\"
+                       \"c25ee04b-...\"
+                       [[\"sample\" \"NA12878\"] [\"sample\" \"NA12879\"]])
+  "
+  [workspace set-name [[entity-type _] & _ :as entities]]
+  ;; TODO: Should we enforce common entity-type for all entities?
+  (let [columns [entity-type entity-type]
+        rows (for [[_ entity-name] entities] [set-name entity-name])]
+    (->> (util/columns-rows->terra-tsv :membership columns rows)
+         .getBytes
+         (import-entities workspace))))
+
+(defn consolidate-entities-to-set
+  "
+  Optionally consolidates ENTITIES into an entity set.
+  Returns the [Type Name] pair for the resulting entity,
+  or the passed-in entity if it was the only one specified.
+
+  Parameters
+  ----------
+    workspace - Fully-qualified Terra Workspace in which the entity set
+                would be created
+    entities  - List of entity [Type Name] pairs
+
+  Example
+  -------
+    (consolidate-entities-to-set \"workspace-namespace/workspace-name\"
+                                 [[\"sample\" \"NA12878\"] [\"sample\" \"NA12879\"]])
+  "
+  [workspace [entity & maybe-entities :as entities]]
+  ;; TODO: Should we enforce common entity-type for all entities?
+  (if (seq maybe-entities)
+    (let [id (str (UUID/randomUUID)) ; TODO: do we have access to a workload-affiliated UUID?
```

Comment:
Flagging this: currently generating a random UUID to serve as a unique identifier for a new entity set entry.  While this works it doesn't reveal much about when / how the entity set entry was uploaded.  Is there some other identifier we can use, perhaps a UUID affiliated with the workload?